getnacelle/nacelle-js
# feat(commerce-queries): add product method

## Why are these changes introduced?

Addresses [ENG-8324](https://nacelle.atlassian.net/browse/ENG-8324) 



## What is this pull request doing?

1. Creates the product query and a number of fragments
2. Updates `codegen.yaml` to have `dedupeFragments: true` so that the query is generated with a single fragment
3. Adds the products method and associated tests


## How to Test

1. Checkout this branch `mdarrik/ENG-8324-add-product-method`
2. Run `npm coverage` typechecking and all tests should succeed
3. Run `npm run build` should build successfully
4. Create an instance of the sdk with the CommerceQueries plugin and use the `sdk.products()` method - should fetch successfully



## Checklist



[ENG-8324]: https://nacelle.atlassian.net/browse/ENG-8324?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
